### PR TITLE
Run tests on Travis CI

### PR DESCRIPTION
**Pull request by spladug**
Exactly what it says on the tin. This does a couple of fixups to make a travis-like build environment happy, then adds a travis config to actually do the build.

### CODE REVIEW

**Comment by spladug** [comment]
A bunch of tests are failing with

> AttributeError: 'functools.partial' object has no attribute 'func_defaults'

which appears to be a bug introduced in recent versions of Mock: testing-cabal/mock#323

**Comment by spladug** [comment]
Ready to go!

:eyeglasses: @KeyserSosa @bsimpson63

**Comment by bsimpson63** [comment]
Anything to worry about with the ImportErrors during the sphinx step?

File: docs/conf.py [hunk 1]
```diff
@@ -3,8 +3,14 @@
 # This file is execfile()d with the current directory set to its
 # containing dir.
 
+import os
+import sys
+
 import alabaster
 
+# where to find the modules to import for autodoc
+sys.path.insert(0, os.path.abspath(".."))
```
Comment (KeyserSosa) [inline docs/conf.py]: *Siiiiiigggghhhhhhh*  

Yup.  Python.  It checks out.

Comment (spladug) [inline docs/conf.py]: Heh yup. It's from the sphinx quickstart too :(

**Comment by spladug** [comment]
@bsimpson63 no, the import errors just mean it can't autodocument the classes from external things. I'll want to fix that up when I get travis doing the documentation builds automatically, but it's not a biggy for now.

**Comment by bsimpson63** [comment]
:fish:

**Comment by KeyserSosa** [comment]
:fish:

**Pull request merged**